Assets/Python/DancingHoskuld/BarbarianDiplomacy.py

```
@@ -1,58 +1,91 @@
+_GAME = CyGame()
+
+# Pre-cache frequently used methods as direct references
+# This eliminates attribute lookup overhead on each call
+_isOption = _GAME.isOption
+_getInfoTypeForString = _GC.getInfoTypeForString
+_debug = BugUtil.debug
+
+# Pre-cache game option type to avoid repeated lookups
+# Python 2.4 doesn't have conditional expressions, so we check at module init
+try:
+    _GAMEOPTION_NO_GOODY_HUTS = GameOptionTypes.GAMEOPTION_MAP_NO_GOODY_HUTS
```

Ruff (0.12.2)

`GameOptionTypes` may be undefined, or defined from star imports

(F405)